lib/ui/painting.dart

```diff
@@ -2855,11 +2858,49 @@ class ImageFilter {
       : assert(matrix4 != null), // ignore: unnecessary_null_comparison
         _data = Float64List.fromList(matrix4),
         _filterQuality = filterQuality,
+        _colorFilter = null,
+        _innerFilter = null,
+        _outerFilter = null,
         _type = _kTypeMatrix {
     if (matrix4.length != 16)
       throw ArgumentError('"matrix4" must have 16 entries.');
   }
 
+  /// Creates an image filter from a [ColorFilter].
+  ///
+  /// The resulting filter is functionally equivalent to the original
+  /// [ColorFilter], but can be used and composed like an [ImageFilter]. For
+  /// example, the [BackdropFilter] widget applies an [ImageFilter] to the
+  /// painted content. To use a [ColorFilter] in a [BackdropFilter], first
+  /// convert the [ColorFilter] to an [ImageFilter] using
+  /// [ImageFilter.fromColorFilter].
+  ///
+  /// A [ColorFilter] may be invalid (`ColorFilter.mode(null, null)` for
```

Comment:
I find it difficult to define "invalid" here (`_toNativeColorFilter` is private).

Reply:
Can we rely on ColorFilter to specify and/or detect that?

Reply:
Now it works with "null" `ColorFilter`s